plugins/vbe/README.md

```diff
@@ -0,0 +1,326 @@
+         status = "completed";
+     };
+```
+
+The fields are as follows:
+
+* compatible - the compatible string for VBE
+* vbe-driver - the actual VBE method being used
+* last-update - information about the last update
+* finish-time - the unix time of the last update, i.e. seconds since 1970, as a
+  64-bit value (0 if no previous update)
+* cur-version - the current installed version, missing if there is none or it is
+  not known
+* status - status of the update. The only supported value is 'completed'
+
+## Adding new VBE methods
+
+New VBE methods can be added in the plugin file `fu-plugin-vbe.c` in the
+`driver_list` array. See the documentation there for how to do this.
+
+## Finding out more
+
+See [fwupd site](https://github.com/fwupd/fwupd) for information about how
+to discuss or chat about fwupd.
+
+You can use the U-Boot mailing list or `u-boot` IRC on `libera.chat` to ask
+questions specific to VBE and firmware.
+
+## Sending patches
+
+Use the [fwupd developer site](https://fwupd.org/lvfs/docs/developers) to find
+information about downloading the code, submitting bugs/feature requests and
+sending patches.
+
+## Vendor ID Security
+
+This does not update USB devices and thus requires no vendor ID set.
+
+## External Interface Access
+
+This plugin requires access to system firmware, e.g. via a file or an eMMC
+device.
+
+## Documentation
+
+The following documents may help in understanding VBE:
+
+* [VBE](https://docs.google.com/document/d/e/2PACX-1vRwxXnft4o4kL1rLJzm8kwsjNZZC1CQcozNFbS_bk-0pV6uejJaVlJwg9kfFeLCEzA6yBlw6eDmdS9t/pub)
+* [VBE Bootflows](https://docs.google.com/document/d/e/2PACX-1vQF7g5vjaoDzRDPWmbYOoOaNQsaPYDaCT0I2LZRNFIT0lD1hqANOcBmMicM-MACFQIjvE_VN9cSvCzy/pub)
+* [VBE Firmware update](https://docs.google.com/document/d/e/2PACX-1vTMxDQ_CsT76vgpMUfxO6YehiWdP03pKS0L6I5DXYEUnZGuFiOubZiAczFAcefT0uSnbZC84fl9nGuP/pub)
+* [FIT](https://github.com/u-boot/u-boot/blob/master/doc/uImage.FIT/source_file_format.txt)
+* [U-Boot Standard boot](https://u-boot.readthedocs.io/en/latest/develop/bootstd.html)
+
+## Useful information
+
+For development you may find the following useful.
+
+To create a link to the VBE plugin for development:
+
+```bash
+    cd /usr/local/lib/x86_64-linux-gnu
+    ln -s /path/to/fwupd/build/plugins/vbe/ fwupd-plugins-6
+```
+
+To set up the vbe directory:
+
+```bash
+    mkdir /var/local/lib/fwupd/vbe
+    chmod a+rwx /var/local/lib/fwupd/vbe
+    dtc /path/to/fwupd/plugins/vbe/example/test.dts -o \
+       /var/local/lib/fwupd/vbe/system.dtb
+```
+
+Some things you might need to install:
+
+```bash
+   sudo apt install build-essential git python3-pip pkg-config cmake \
+ libglib2.0-dev liblzma-dev python3.10-venv libgirepository1.0-dev \
+ gtk-doc-tools libgnutls28-dev libgpgme-dev gnutls-bin valac \
+ libusb-1.0-0-dev python3-jinja2 python3-toml python3-typogrify \
+ gcab gettext ninja-build libjson-glib-dev libxml2-utils libfdt-dev \
+ gir1.2-gusb-1.0 appstream-util
```

Comment:
I would hope that none of this is necessary.  Shouldn't `./contrib/setup` do all this for you?  If not - some stuff is missing in `contrib/ci/dependencies.xml`.

Reply:
I get this when running that script:

E: Unable to locate package gi-docgen

so I had to do various things manually

This is Ubuntu 20.04.4 LTS

Reply:
@superm1 is there any way to tell apt-get to ignore packages it knows nothing about? I don't really want to create an ubuntu-old-lts variant in deps.xml

Reply:
Not that I can tell.  Two ideas:
1. We add a new optional field to dependencies.xml that is "minimum OS version" to install this dep?  Then the helper script can examine this extra field and match to lsb_release to decide.
2. Helper script can use python3-apt to match whether or not the package is plausible to install or not before populating the install command.

Reply:
I think that's too antisocial; Chromium dev environment is still 20.04 for now.
Here's a stab at 2: https://github.com/fwupd/fwupd/pull/4748

Reply:
@sjg20 that was merged.  If you rebase on latest main, you shouldn't encounter that on Ubuntu 20.04.

Reply:
(and use ./contrib/migrate.py -- then fix up any issues -- we broke internal ABI a bit)

Reply:
Yes, fixed, thanks!

Now I am getting an error with progress. Please see #4749